### DISCUSSION TITLE
ng-packagr rootDir issues

### DISCUSSION THREAD

**Discussion by kainiedziela** [NG-ZORRO/ng-zorro-antd#5974] [Q&A]
Hello ng-zorro team! Big fan!

I'm working on creating an angular library myself and have to into an issue that you guys solved, however I cannot reverse engineer the exact way you did this.

I want to have a component structure similiar to yours and that of the @angular/material package.

```TS
import { ButtonModule } from '@package/button`; // like this

import { ButtonModule } from '@package`; // not like this
```

And while the overall library structure supports this when I want to use some shared functions the way you guys use your `core` directory, ng-packagr will not build the component, because the shared directory is not in the root directory of a given sub-library.

Can you please point me in the direction of how you fixed this?

**Comment by stygian-desolator** [comment]
`@package/core` is also a module similar to `@package/button`
